core/src/main/java/org/infinispan/commands/functional/ReadWriteKeyCommand.java

```diff
@@ -130,4 +128,26 @@ public void init(ComponentRegistry componentRegistry) {
          ((InjectableComponent) f).inject(componentRegistry);
       }
    }
+
+   @Override
+   public final boolean isReturnValueExpected() {
+      return true;
+   }
+
+   @Override
+   public Metadata getMetadata() {
```

Comment:
@rvansa I'm patient, I just prefer "done" rather than "perfect". I suggest to do :

1. Throw unsupported operation exception for SC in the case of ReadWriteKVCommand
2. Delete the MetaAwareX and cherry-pick the code you pointed if necessary
3. Create an new issue to specifically treat the concern about missing value during early rebalance : this touches merge but compute methods which are already in 9.1 are concerned, so it deserves a new issue. I can treat this issue by myself, I just need some support to reproduce the problem as I couldn't last week and I got really fed up for a while 😢 


WDYT ?